sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -250,6 +252,8 @@ class Dataset[T] private[sql](
     val hasMoreData = takeResult.length > numRows
     val data = takeResult.take(numRows)
 
+    lazy val timeZone = TimeZone.getTimeZone(sparkSession.sessionState.conf.sessionLocalTimeZone)
```

Comment:
This will be used not once but num rows * num cells of TimestampType. I think we should leave it as is.